packages/frontend/src/components/user-auth-form.tsx

```diff
@@ -8,14 +8,32 @@ import { Label } from '@/components/ui/label';
 import { Icons } from '@/components/icons';
 import Link from 'next/link';
 import { usePathname } from 'next/navigation';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useForm, SubmitHandler } from 'react-hook-form';
+import { userAuthSchema } from '@/lib/validation/auth';
+
+type FormSchema = z.infer<typeof userAuthSchema>;
 
 export function UserAuthForm() {
   const pathName = usePathname();
   const isRegister = pathName === '/register';
 
+  const {
+    register,
+    handleSubmit,
+    formState: { errors }
+  } = useForm<FormSchema>({
+    resolver: zodResolver(userAuthSchema)
+  });
+
+  const onSubmit: SubmitHandler<FormSchema> = (data) => {
+    console.log(data);
+  };
+
   return (
     <div className={cn('grid gap-6')}>
-      <form>
+      <form onSubmit={handleSubmit(onSubmit)}>
```

Comment:
**nitpick (non-blocking):** For future forms, we could use the ShadCN Ui and RHF component here https://ui.shadcn.com/docs/components/form. Does same as the current implementation with already designed form components making it easier to implement.